tests/model/eventsources/test_sns_event_source.py

```diff
@@ -77,3 +78,20 @@ def test_to_cloudformation_when_sqs_subscription_disable(self):
         self.assertEqual(subscription.Endpoint, "arn:aws:lambda:mock")
         self.assertIsNone(subscription.Region)
         self.assertIsNone(subscription.FilterPolicy)
+
+    def test_to_cloudformation_when_region_is_optin(self):
```

Comment:
We should define what happens when a region is not returned at all by `boto3`, and then test that behavior holds in the same way.